controllers/profiles/getOtherProfileByUsername.js

```diff
@@ -3,12 +3,29 @@ const {checkMoreOrLess} = require('../../helpers/checkMoreOrLess');
 
 module.exports = async (req, res) => {
   try {
+    // check this route is use auth or not
+    let excludeField = [
+      'human_id',
+      'is_backdoor_user',
+      'encrypted',
+      'created_at',
+      'updated_at',
+      'real_name',
+      'last_active_at',
+      'status',
+      'is_banned'
+    ];
+    if (req?.userId) {
+      excludeField = ['human_id', 'is_backdoor_user', 'encrypted'];
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for excluding fields based on the presence of `userId` is correct. However, it's not efficient to declare the `excludeField` array twice. You can improve this by declaring a base array and then adding additional fields if `userId` is present.
```diff
-    let excludeField = [
-      'human_id',
-      'is_backdoor_user',
-      'encrypted',
-      'created_at',
-      'updated_at',
-      'real_name',
-      'last_active_at',
-      'status',
-      'is_banned'
-    ];
-    if (req?.userId) {
-      excludeField = ['human_id', 'is_backdoor_user', 'encrypted'];
-    }
+    const baseFields = ['human_id', 'is_backdoor_user', 'encrypted'];
+    const additionalFields = ['created_at', 'updated_at', 'real_name', 'last_active_at', 'status', 'is_banned'];
+    const excludeField = req?.userId ? baseFields : [...baseFields, ...additionalFields];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->